File: examples/postcode/lookup/node/lookupPostcode.js
const { PostcodeService } = require("m3o/postcode");

// Lookup a postcode to retrieve the related region, county, etc
async function lookupPostcode() {
  let postcodeService = new PostcodeService(process.env.MICRO_API_TOKEN);
  let rsp = await postcodeService.lookup({
    postcode: "SW1A 2AA",
  });
  console.log(rsp);
}

lookupPostcode();

File: examples/postcode/lookup/go/lookupPostcode.go
package example

import (
	"fmt"
	"github.com/micro/services/clients/go/postcode"
	"os"
)

// Lookup a postcode to retrieve the related region, county, etc
func LookupPostcode() {
	postcodeService := postcode.NewPostcodeService(os.Getenv("MICRO_API_TOKEN"))
	rsp, err := postcodeService.Lookup(&postcode.LookupRequest{
		Postcode: "SW1A 2AA",
	})
	fmt.Println(rsp, err)
}
